tcn_hpl/callbacks/plot_metrics.py

```diff
@@ -362,6 +364,40 @@ def on_test_epoch_end(
         test_acc = pl_module.test_acc.compute()
         test_f1 = pl_module.test_f1.compute()
 
+        # Create activity predictions KWCOCO JSON
+        truth_dset_fpath = trainer.datamodule.hparams["coco_test_activities"]
+        truth_dset = kwcoco.CocoDataset(truth_dset_fpath)
+        acts_dset = kwcoco.CocoDataset()
+        acts_dset.fpath = self._preds_dset_output_fpath
+        acts_dset.dataset['videos'] = truth_dset.dataset['videos']
+        acts_dset.dataset['images'] = truth_dset.dataset['images']
+        acts_dset.index.build(acts_dset)
+        acts_dset.dataset['categories'] = truth_dset.dataset['categories']
```

Comment:
I'd still swap the order of these.
```suggestion
        acts_dset.dataset['categories'] = truth_dset.dataset['categories']
        acts_dset.index.build(acts_dset)
```